### PR TITLE
Feature/kyv 847 subscription renewal error monitoring

### PR DESCRIPTION
**Pull request by laukkanenj**

### CODE REVIEW

File: messageapi/src/main/java/fi/hel/verkkokauppa/message/dto/MessageDto.java [hunk 1]
```diff
@@ -24,7 +24,6 @@ public class MessageDto extends BaseDto {
     private String sender;
 
     @NotEmpty
-    @Email
```
Comment (severikupari1) [inline messageapi/src/main/java/fi/hel/verkkokauppa/message/dto/MessageDto.java:27]: Why did you remove this? Is there a validation for multiple emails?

File: common/src/main/java/fi/hel/verkkokauppa/common/queue/service/SendNotificationService.java [hunk 2]
```diff
@@ -95,4 +94,37 @@ public void sendRefundMessageNotification(
...
+    ) {
+        sendErrorNotification(
+                endpoint,
+                message + " " + exception.getMessage(),
+                exception.getStackTrace().toString()
+        );
+    }
+
+    public void sendErrorNotification(
+            String endpoint,
```
Comment (severikupari1) [inline common/src/main/java/fi/hel/verkkokauppa/common/queue/service/SendNotificationService.java:110]: This can be function also etc? message should be one string that is passed to this function. Makes this more usable in different error notification places.

File: common/src/main/java/fi/hel/verkkokauppa/common/queue/service/SendNotificationService.java [hunk 3]
```diff
@@ -95,4 +94,37 @@ public void sendRefundMessageNotification(
...
+            String message,
+            String cause
+    ) {
+        String toQueue = queueConfigurations.getErrorEmailNotificationsQueue();
+        try {
+            ActiveMQQueue queue = new ActiveMQQueue(toQueue);
+            ErrorMessage queueMessage = ErrorMessage.builder()
+                    .eventType(EventType.ERROR_EMAIL_NOTIFICATION)
+                    .eventTimestamp(DateTimeUtil.getDateTime())
+                    .message("Endpoint: " + endpoint + ". " + message)
```
Comment (severikupari1) [inline common/src/main/java/fi/hel/verkkokauppa/common/queue/service/SendNotificationService.java:120]: these could come just from one string, sender can create with string builder if wanted etc using parameters there.

File: common/src/main/java/fi/hel/verkkokauppa/common/queue/service/SendNotificationService.java [hunk 4]
```diff
@@ -95,4 +94,37 @@ public void sendRefundMessageNotification(
...
     }
 
+    public void sendErrorNotification(
+            String endpoint,
+            String message,
+            Exception exception
+    ) {
+        sendErrorNotification(
+                endpoint,
+                message + " " + exception.getMessage(),
```
Comment (severikupari1) [inline common/src/main/java/fi/hel/verkkokauppa/common/queue/service/SendNotificationService.java:104]: Make endpoint and message to be one message parameter no need for duplicate strings.

File: orderapi/src/main/java/fi/hel/verkkokauppa/order/api/admin/SubscriptionAdminController.java [hunk 5]
```diff
@@ -297,7 +313,7 @@ public ResponseEntity<List<String>> checkExpiringCards() {
 
     /**
      * "Get all active subscriptions that are expiring in the next X days."
-     *
+     * <p>
```
Comment (severikupari1) [inline orderapi/src/main/java/fi/hel/verkkokauppa/order/api/admin/SubscriptionAdminController.java:316]: Is this typo?

File: messageapi/src/test/java/fi/hel/verkkokauppa/message/service/MessageServiceTest.java [hunk 6]
```diff
@@ -95,14 +96,54 @@ public void sendOrderConfirmationEmail() throws Exception {
 
         // Verify sent email values
         List<SmtpMessage> emails = dumbster.getReceivedEmails();
+        dumbster.close();
```
[the remark this answers is not in the document]
Comment (laukkanenj) [reply messageapi/src/test/java/fi/hel/verkkokauppa/message/service/MessageServiceTest.java:99]: This was easiest way to get dumbster working for multiple tests. New one could not be started unless previous was closed and having dumbster on class level threw other errors.

File: common/src/main/java/fi/hel/verkkokauppa/common/queue/service/SendNotificationService.java (hunk 4 above)
Comment (laukkanenj) [reply common/src/main/java/fi/hel/verkkokauppa/common/queue/service/SendNotificationService.java:104]: Made them separate so that it would be clear that endpoint should be given but can be combined

File: orderapi/src/main/java/fi/hel/verkkokauppa/order/api/admin/SubscriptionAdminController.java (hunk 5 above)
Comment (laukkanenj) [reply orderapi/src/main/java/fi/hel/verkkokauppa/order/api/admin/SubscriptionAdminController.java:316]: probably

File: eventsapi/src/main/java/fi/hel/verkkokauppa/events/notification/ErrorEmailNotificationListener.java [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+import javax.jms.JMSException;
+import javax.jms.TextMessage;
+import java.io.IOException;
+import java.util.UUID;
+
+@Component
+@Slf4j
+public class ErrorEmailNotificationListener extends BaseEmailNotifier<ErrorMessage> {
+
+    @Value("${error.notification.email:jarkko.amb@gmail.com}")
```
Comment (severikupari1) [inline eventsapi/src/main/java/fi/hel/verkkokauppa/events/notification/ErrorEmailNotificationListener.java:26]: Change default value to be correct/ remove it.

File: eventsapi/src/test/java/fi/hel/verkkokauppa/events/notification/ErrorEmailNotificationListenerTest.java [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+    @Autowired
+    SendNotificationService sendNotificationService;
+
+    @Autowired
+    QueueConfigurations queueConfigurations;
+
+    @Autowired
+    private RestServiceClient restServiceClient;
+
+    @Value("${error.notification.email:jarkko.amb@gmail.com}")
```
Comment (severikupari1) [inline eventsapi/src/test/java/fi/hel/verkkokauppa/events/notification/ErrorEmailNotificationListenerTest.java:42]: Change this to some other

File: messageapi/src/test/java/fi/hel/verkkokauppa/message/service/MessageServiceTest.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (severikupari1) [reply messageapi/src/test/java/fi/hel/verkkokauppa/message/service/MessageServiceTest.java:99]: Oh ok!

[the remark this answers is not in the document]
Comment (severikupari1) [reply messageapi/src/test/java/fi/hel/verkkokauppa/message/service/MessageServiceTest.java:99]: Could this be done in afterEach function?

File: common/src/main/java/fi/hel/verkkokauppa/common/queue/service/SendNotificationService.java (hunk 4 above)
Comment (severikupari1) [reply common/src/main/java/fi/hel/verkkokauppa/common/queue/service/SendNotificationService.java:102]: But is there allways an endpoint what is called?

**Review by severikupari1 — APPROVED** [review]
Good job! That setup function was good thing to do in test.
